### DISCUSSION TITLE
Home/list page doesn't reflect changes to meetings, doesn't search using pulldowns

### DISCUSSION THREAD

**Discussion by UpperValleyAA** [code4recovery/12-step-meeting-list#447] [Q&A] [answered]
Hi! Super grateful for this wonderful plug-in.

However, for some few weeks I have had a few problems with the plug-in's "home" or meeting-list page (due to WP update?).

1) When I update a detail in Dashboard/Meetings/Edit Meeting, the new detail appears in that meeting's individual page, but not on the meeting-list page. E.g., the Sunday 7:30pm meeting "[Twin State Group Speaker Meeting](https://uppervalleyaa.com/meetings/twin-state-group-speaker-meeting)" has the correct address on its page, but an incorrect (old) address on uppervalleyaa.com, the meeting list page.
   I have tried creating new "test" meetings. Their pages work---  e.g., https://uppervalleyaa.com/meetings/test --- but they don't appear on the uppervalleyaa.com list page. 
    
2) The pulldown sorting menus on uppervalleyaa.com have stopped working, e.g., pulling down "Tuesday" has no result.

I tried deactivating all plug-ins, turning them on again. Tried saving all permalinks. 

Trying to eliminate all possibilities that might be affecting the list page, I attempted to delete the FAQ-suggested customizing codes
_// Make meeting page front or home page.
add_action('pre_get_posts', 'tsml_front_page');
// Make any day the default day of week.
$tsml_defaults['day'] = null;_
from the functions.php using Theme Editor (which was used to add them; the site worked fine with them, but I was trying everything) caused the following error message:
_Unable to communicate back with site to check for fatal errors, so the PHP change was reverted. You will need to upload your PHP file change by some other means, such as by using SFTP_

Checking the php with the PHP COMPATABILITY plug-in gets message that 12 Step Meeting List plugin may not be compatible, with 1 error and 1 warning:
  _143 | ERROR   | Function set_magic_quotes_runtime() is deprecated since PHP 5.3 and removed since PHP 7.0
 1934 | WARNING | INI directive 'safe_mode' is deprecated since PHP 5.3 and removed since PHP 5.4_

I contacted my hosting service (1&1 Ionos) and they ran diagnostics, found no issues other than something wrong with the plug-in---   but they couldn't define or fix that and said I needed to bring it to the plug-in team's attention.

That's it! This is way beyond me. Any help would be MUCH appreciated.

Bill for UpperValleyAA

 

.

**Comment by brianw-area24** [comment, marked as answer]
Hi Bill,

Thanks you for the detailed description of the problem, and including the URL to your site, that helped a lot.

It sounds like you may have 2 separate issues, but they both seem to be permissions related. I'll address them separately below.

First, when you edit a meeting, and the change shows on the details page, but not on the listing page. The TSML keeps a cache file with meeting information. When it gets the meetings for the list of meetings page, it reads this file, if it's available, which saves it from having to read all that information from the database. This cache file is supposed to get re-written every time you update a meeting. It sounds like this file isn't being rewritten, so the list page is reading old information from this cache file. Here's what I'd like you to do for testing:
- Go to the WordPress Dashboard
- Edit a meeting
- Click on Update button, you don't need to change anything, just click Update
- Check the date on the cache file, it's located at wp-content/tsml-cache-<unique-code-here>.json 

After doing the update, this file should have gotten rewritten, so have a recent update date. My guess, yours will have an old update date. If so, it means that your web server doesn't have permission to write to this file anymore. Either the file permissions have changed, the owner of the file has changed, or something else has changed causing these permission issues.

Likewise, I think the second issue, the dropdown lists not working, is also a permissions issue. I say this because, I turned on the web console on my browser. Then I went to your meeting list, and selected Tuesday, and the console showed an error of "POST https://uppervalleyaa.com/wp-admin/admin-ajax.php [HTTP/2 401 Unauthorized 154ms]" See the attached image. This indicates to me that the web server isn't able to use the admin-ajax.php file.

My suggestion would be to go back to your host provider, with this additional information, and see if they can find something. Hopefully they will.

Brian W.

![Screen Shot 2021-07-14 at 12 59 33 AM](https://user-images.githubusercontent.com/55821195/125570741-5df11445-2862-463f-8fc0-15aedab8c184.png)

**Comment by UpperValleyAA** [comment]
Thanks, brianw! I truly appreciate your thoughtful answer.
I'm starting by trying to work on your first suggestion. Please forgive my utter newbie-ism, but I am not certain what you mean by "Check the date on the cache file, it's located at wp-content/tsml-cache-.json." Where do I find that cache file? I tried entering "uppervalleyaa.com/wp-content/tsml-cache-.json" into the browser and got "page not found" result. The Theme Editor shows no tsml-cache-.json that I can find.

**Reply by brianw-area24** [reply]
This might depend on your host provider. Hopefully your provider has a method to show what files are on the file system, many times they use a system called cPanel and it's one of the options in that system.

wp-content is a subfolder/directory within the WordPress folder and the tsml-cache file is a file in that subfolder. However, the filename has a unique code in it, so it's not easy to guess the full filename. (I forgot to include that part in my previous response. On my test site, the filename is tsml-cache-323901f1a6.json, you're will start with tsml-cache-, but the code will be different)

If you can't find it, your host provider should be able to easily get you the modified date for this file.

Brian W

**Comment by UpperValleyAA** [comment]
Thanks, Brian!
The noon meetings were approaching, and folks up here needed the links, so I took the scorched-earth approach to the various problems---   some of which seemed possibly linked to a domain migration back when---  and deleted the WP site, deleted all data from host server, and built a new ugly / quick site. So I will never know whether some of your kind suggestions would have solved thoe problems---   I suspect they would have, but there were other problems behind those...
Now I'm back to newbie issues like, how do I safely edit the appearance of the plug-in-provided meeting list page? It's not one of the pages in Pages, right? So if I want to put a button in the header of all pages, to click to monthly business meeting, do I use Appearance/Customize or Theme Editor or... ?
Anyway, many thanks---
Bill

**Comment by brianw-area24** [comment]
Looks like you've got a pretty good handle on it. Remember, there are a lot of good tutorial/help out there for basic Wordpress information, and Wordpress is pretty easy to get up and running. We are here primarily to help support the 12-step-meeting-list (TSML) plugin and the new TSML-UI, and we're happy to help with any questions along those lines.

Thank you for your service, and keep up the good work.
Brian W
